### PR TITLE
Reverting back the method signature on the KafkaMetricsReporterServic…

### PR DESCRIPTION
**Pull request by joshuaspaulding**
…e constructor.

The KafkaMetricReporterService appears to be broken with the addition of `adminClient` in the method signature.  I removed the call to createTopicIfNotExists and reverted the method sig back to Map<String, Object>, String.

#235

### CODE REVIEW

**Comment by andrewchoi5** [comment]
Hello Joshua, please refer to the excerpt from my log. Also, your PR's CI travis build is currently failing. please check.

The HEAD of the master branch seems to be working fine on my end. Have you tried running 

`./gradlew clean build && ./gradlew jar && ./bin/kafka-monitor-start.sh config/kafka-monitor.properties`

with (of course) at least on kafka server and zookeeper server running?

```[2020-04-09 15:44:17,184] INFO ==============================================================
kmf:type=kafka-monitor:offline-runnable-count=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:produce-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-avg=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-total=776.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-consumed-total=754.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-duplicated-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-delay-ms-avg=3.282493368700265
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-rate=12.970949085681811
kmf.services:name=single-cluster-monitor,type=produce-service:produce-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-total=10.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-avg=2.7
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-max=11.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-99th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-9999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-total=0.0
 (com.linkedin.kmf.services.DefaultMetricsReporterService)
[2020-04-09 15:44:18,184] INFO ==============================================================
kmf:type=kafka-monitor:offline-runnable-count=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:produce-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-avg=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-total=794.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-consumed-total=772.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-duplicated-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-delay-ms-avg=3.2629533678756477
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-rate=13.272265311580638
kmf.services:name=single-cluster-monitor,type=produce-service:produce-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-total=11.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-avg=2.6363636363636362
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-max=11.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-99th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-9999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-total=0.0
 (com.linkedin.kmf.services.DefaultMetricsReporterService)```

File: src/main/java/com/linkedin/kmf/services/KafkaMetricsReporterService.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
   private final String _topic;
   private final ObjectMapper _parser = new ObjectMapper();
 
-  public KafkaMetricsReporterService(Map<String, Object> props, String name, AdminClient adminClient) throws Exception {
+  public KafkaMetricsReporterService(Map<String, Object> props, String name) throws Exception {
```
Comment (andrewchoi5) [inline src/main/java/com/linkedin/kmf/services/KafkaMetricsReporterService.java:43]: Hello Joshua, please refer to the excerpt from my log. Also, your PR's CI travis build is currently failing. please check.

The HEAD of the master branch seems to be working fine on my end. Have you tried running 

`./gradlew clean build && ./gradlew jar && ./bin/kafka-monitor-start.sh config/kafka-monitor.properties`

with (of course) at least on kafka server and zookeeper server running?

```[2020-04-09 15:44:17,184] INFO ==============================================================
kmf:type=kafka-monitor:offline-runnable-count=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:produce-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-avg=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-total=776.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-consumed-total=754.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-duplicated-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-delay-ms-avg=3.282493368700265
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-rate=12.970949085681811
kmf.services:name=single-cluster-monitor,type=produce-service:produce-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-total=10.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-avg=2.7
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-max=11.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-99th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-9999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-total=0.0
 (com.linkedin.kmf.services.DefaultMetricsReporterService)
[2020-04-09 15:44:18,184] INFO ==============================================================
kmf:type=kafka-monitor:offline-runnable-count=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:produce-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-availability-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-avg=1.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-avg=0.0
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-total=794.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-consumed-total=772.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-lost-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-duplicated-total=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:records-delay-ms-avg=3.2629533678756477
kmf.services:name=single-cluster-monitor,type=produce-service:records-produced-rate=13.272265311580638
kmf.services:name=single-cluster-monitor,type=produce-service:produce-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=consume-service:consume-error-rate=0.0
kmf.services:name=single-cluster-monitor,type=commit-availability-service:offsets-committed-total=11.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-avg=2.6363636363636362
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-max=11.0
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-99th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-latency-service:commit-offset-latency-ms-9999th=10.995601759296282
kmf.services:name=single-cluster-monitor,type=commit-availability-service:failed-commit-offsets-total=0.0
 (com.linkedin.kmf.services.DefaultMetricsReporterService)```

Comment (joshuaspaulding) [reply src/main/java/com/linkedin/kmf/services/KafkaMetricsReporterService.java:43]: Yes, the DefaultMetricsReporterService does work for me as well. My issue and PR is with regards to the KafkaMetricsReporterService.

**Comment by andrewchoi5** [comment]
@joshuaspaulding Hi Joshua, is this still active?

**Comment by joshuaspaulding** [comment]
No, this can be closed.

**Comment by andrewchoi5** [comment]
https://github.com/linkedin/kafka-monitor/pull/252

**Pull request closed without merging**